src/goal.cpp

```diff
@@ -41,29 +48,39 @@ auto make_frame_tests(std::vector<Eigen::Isometry3d> goal_frames,
     return tests;
 }
 
-auto make_pose_cost_fn(Eigen::Isometry3d goal, size_t goal_link_index, double rotation_scale)
-    -> PoseCostFn {
+auto make_pose_cost_fn(Eigen::Isometry3d goal,
+                       size_t goal_link_index,
+                       double position_scale,
+                       double rotation_scale) -> PoseCostFn {
+    if (position_scale > 0.0) {
+        if (rotation_scale > 0.0) {
+            return [=](std::vector<Eigen::Isometry3d> const& tip_frames) -> double {
+                auto const& frame = tip_frames[goal_link_index];
+                return std::pow(linear_distance(goal, frame) * position_scale, 2) +
+                       std::pow(angular_distance(goal, frame) * rotation_scale, 2);
+            };
+        }
+        return [=](std::vector<Eigen::Isometry3d> const& tip_frames) -> double {
+            auto const& frame = tip_frames[goal_link_index];
+            return std::pow(linear_distance(goal, frame) * position_scale, 2);
+        };
+    }
     if (rotation_scale > 0.0) {
```

Comment:
This is kind of a nit, since early returns are doing it already, and it was already like this before your changes... but could we actually explicitly include `else` statements? So

```cpp
if (position_scale > 0.0) {
    if (rotation_scale > 0.0) {
        ...
    } else {
        ...
    }
} else if (rotation_scale > 0.0) {
    ...
} else {
    ...
}
```

Reply:
Yeah, that bothered me too. I just did it like this to keep with the previous state, but I also think its clearer with explicit `else`.